modules/openthread/platform/diag.c

```diff
@@ -291,25 +319,152 @@ otError otPlatDiagGpioGetMode(uint32_t aGpio, otGpioMode *aMode)
 
 static otError startModCarrier(otInstance *aInstance, uint8_t aArgsLength, char *aArgs[])
 {
-	ARG_UNUSED(aInstance);
-	ARG_UNUSED(aArgsLength);
-
 	bool enable = true;
 	uint8_t data[OT_RADIO_FRAME_MAX_SIZE + 1];
 
 	if (aArgsLength <= 0) {
 		return OT_ERROR_INVALID_ARGS;
 	}
 
+	if (!otPlatDiagModeGet() && (sTransmitMode != diag_transmit_mode_idle ||
+				     sTransmitMode != diag_transmit_mode_modcarrier)) {
+		return OT_ERROR_INVALID_STATE;
+	}
+
 	if (strcmp(aArgs[0], "stop") == 0) {
 		enable = false;
+		sTransmitMode = diag_transmit_mode_idle;
 	} else {
 		if (hex2bin(aArgs[0], strlen(aArgs[0]), data, ARRAY_SIZE(data)) == 0) {
 			return OT_ERROR_INVALID_ARGS;
 		}
+		sTransmitMode = diag_transmit_mode_modcarrier;
 	}
 
 	return platformRadioTransmitModulatedCarrier(aInstance, enable, data);
 }
 
 #endif
+
+void otPlatDiagAlarmCallback(otInstance *aInstance)
+{
+	uint32_t now;
+	otRadioFrame *sTxPacket;
```

Comment:
```suggestion
	otRadioFrame *txPacket;
```
`sTxPacket` suggests that variable is static